src/broadcast.jl

```diff
@@ -8,199 +9,143 @@ else
     const broadcast_indices = broadcast_shape
 end
 
-if VERSION >= v"0.5.0-dev+5189"
-    _to_shape(dims::Base.DimsOrInds) = map(_to_shape, dims)
-    _to_shape(r::Base.OneTo) = Int(last(r))
+if !isdefined(Base.Broadcast, :ftype) # Julia < 0.6
+    ftype(f, A) = typeof(f)
+    ftype(f, A...) = typeof(a -> f(a...))
+    ftype(T::DataType, A) = Type{T}
+    ftype(T::DataType, A...) = Type{T}
 else
-    _to_shape(x) = x
+    using Base.Broadcast: ftype
 end
 
-if VERSION < v"0.5.0-dev+5434"
-    function gen_nullcheck(narrays::Int, nd::Int)
-        e_nullcheck = macroexpand(:( @nref $nd isnull_1 d->j_d_1 ))
-        for k = 2:narrays
-            isnull = Symbol("isnull_$k")
-            j_d_k = Symbol("j_d_$k")
-            e_isnull_k = macroexpand(:( @nref $nd $(isnull) d->$(j_d_k) ))
-            e_nullcheck = Expr(:||, e_nullcheck, e_isnull_k)
-        end
-        return e_nullcheck
+if !isdefined(Base.Broadcast, :ziptype) # Julia < 0.6
+    if isdefined(Base, :Iterators)
+        using Base.Iterators: Zip2
+    else
+        using Base: Zip2
     end
+    ziptype(A) = Tuple{eltype(A)}
+    ziptype(A, B) = Zip2{Tuple{eltype(A)}, Tuple{eltype(B)}}
+    @inline ziptype(A, B, C, D...) = Zip{Tuple{eltype(A)}, ziptype(B, C, D...)}
```

Comment:
Likewise here, the changes in https://github.com/JuliaLang/julia/pull/19421 might be relevant. Best!